src/datalinker/dependencies/hips.py

```diff
@@ -56,39 +172,91 @@ async def _build_hips_list(
         logger
             Logger for any error messages.
         """
-        lists = []
-        for dataset in HIPS_DATASETS:
-            url = str(config.hips_base_url).rstrip("/") + f"/{dataset}"
-            r = await client.get(
-                url + "/properties",
-                headers={"Authorization": f"bearer {config.token}"},
+        config = config_dependency.config()
+
+        # Determine base URL: use default dataset if configured,
+        # otherwise legacy URL
+        if (
+            config.hips_default_dataset
+            and config.hips_default_dataset in config.hips_datasets
```

Comment:
The second check is enforced by the `Config` class so I think shouldn't be needed here.